programming_examples/shim_dma_2d/README.md

```diff
@@ -0,0 +1,35 @@
+# shim_dma_2d
+
+This example demonstrates how data may be moved using shim DMA operations. Specifically, in this example a 2-dimensional block of data (sometimes referred to in the code as an *image*) is set to have some specific values.
+
+It was transferred to a compute core using DMA. The compute core then modifies an upper corner of the data (sometimes referred to in code as the *tile*).
+
+The logic in this example is defined in [shim_dma_2d.py](shim_dma_2d.py).
+
+For illustrative purposes, there are 3 ways to run and test this example.
+
+## Run and test with AIR utility functions
+
+This is the cleanest and simplest method of running MLIR-AIR code on NPUs, and uses code in the [run.py](run.py) file.
```

Comment:
This might be a little generic, it doesn't really describe what is happening: compile + run/verify

Reply:
Good point! I elaborated quite a bit in a recent commit. Can you double check that what I wrote is precise/correct/understandable?

I wasn't sure how much to put for explanation, as I think this is just an example and not part of a programming guide. But maybe that thinking is wrong?